pkg/operator/staticpod/certsyncpod/certsync_controller.go

```diff
@@ -198,11 +198,14 @@ func (c *CertSyncController) sync() error {
 			}
 
 			// remove missing content
-			if err := os.RemoveAll(getSecretDir(c.destinationDir, s.Name)); err != nil {
-				c.eventRecorder.Warningf("CertificateUpdateFailed", "Failed removing file for secret: %s/%s: %v", secret.Namespace, secret.Name, err)
-				errors = append(errors, err)
+			secretFile := getSecretDir(c.destinationDir, s.Name)
+			if _, err := os.Stat(secretFile); !os.IsNotExist(err) {
+				if err := os.RemoveAll(secretFile); err != nil {
+					c.eventRecorder.Warningf("CertificateUpdateFailed", "Failed removing file for missing secret: %s/%s: %v", secret.Namespace, secret.Name, err)
+					errors = append(errors, err)
```

Comment:
I think that we have to `continue` otherwise we will fire `CertificateRemoved.

Reply:
i inverted the logic here and checking if the file exists before deleting it in separate `if`.